code/modules/mapping/verify.dm

```diff
@@ -0,0 +1,97 @@
+/// An error report generated by [parsed_map/check_for_errors].
+/datum/map_report
+    var/original_path
+    var/list/bad_paths = list()
+    var/list/bad_keys = list()
+    /// Whether this map can be loaded safely despite the errors.
+    var/loadable = TRUE
+    var/crashed = TRUE
+
+    var/static/tag_number = 0
+
+/datum/map_report/New(datum/parsed_map/map)
+    original_path = map.original_path || "Untitled"
+
+/// Show a rendered version of this report to a client.
+/datum/map_report/proc/show_to(client/C)
+    var/list/html = list()
+    html += "<p>Report for map file <tt>[original_path]</tt></p>"
+    if(crashed)
+        html += "<p><b>Validation crashed</b>: check the runtime logs.</p>"
+    if(!loadable)
+        html += "<p><b>Not loadable</b>: some tiles are missing their turfs or areas.</p>"
+
+    if(bad_paths.len)
+        html += "<p>Bad paths: <ol>"
+        for(var/path in bad_paths)
+            var/list/keys = bad_paths[path]
+            html += "<li><tt>[path]</tt>: used in ([keys.len]): <tt>[keys.Join("</tt>, <tt>")]</tt>"
+        html += "</ol></p>"
+
+    if(bad_keys.len)
+        html += "<p>Bad keys: <ul>"
+        for(var/key in bad_keys)
+            var/list/messages = bad_keys[key]
+            html += "<li><tt>[key]</tt>"
+            if(messages.len == 1)
+                html += ": [bad_keys[key][1]]"
+            else
+                html += "<ul><li>[messages.Join("</li><li>")]</li></ul>"
+            html += "</li>"
+        html += "</ul></p>"
+    C << browse(html.Join(), "window=[tag];size=600x400")
+
+/datum/map_report/Topic(href, href_list)
+    . = ..()
+    if (.)
+        return
+    if (href_list["show"])
```

Comment:
Given these have easy to guess tags it should probably check for some admin permissions.